x-pack/plugin/core/src/main/java/org/elasticsearch/protocol/xpack/watcher/PutWatchRequest.java

```diff
@@ -180,6 +184,12 @@ public ActionRequestValidationException validate() {
         if (ifSeqNo != UNASSIGNED_SEQ_NO && version != Versions.MATCH_ANY) {
             validationException = addValidationError("compare and write operations can not use versioning", validationException);
         }
+
+        if (version != Versions.MATCH_ANY) {
+            DEPRECATION_LOGGER.deprecated(
+                "Usage of internal versioning for optimistic concurrency control is deprecated and will be removed. Please use" +
```

Comment:
Is there no test for watcher? I see that there are REST tests for index/update/delete where you had to suppress this deprecation, but there seems none for watcher?